src/main/java/com/parsing/api/ExtractorLotInformation.java

```diff
@@ -0,0 +1,210 @@
+            } catch (URISyntaxException | JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    @Transactional
+    public void saveLotResult(JsonNode data, String id) {
+        LotResult lotResult = new LotResult();
+        lotResult.setId(id);
+        String status = data.get("status").textValue();
+        if (status.equals(LOT_STATUS)) {
+            extractLotTotalPrice(data, lotResult);
+            extractBuyer(data, lotResult);
+            extractParticipants(data, lotResult);
+            extractDk(data, lotResult);
+            extractLotUrl(data, lotResult);
+            extractPdfUrl(data, lotResult);
+            extractSeller(data, lotResult);
+            extractDateModified(data, lotResult);
+            lotResult.setStatus(Status.CREATED);
+            lotResultRepository.save(lotResult);
+        }
+    }
+
+    private void extractLotTotalPrice(JsonNode data, LotResult lotResult) {
+        JsonNode contracts = data.get("contracts");
+        for (JsonNode contract : contracts) {
+            String price = contract.get("value").get("amount").toString();
+            BigDecimal lotTotalPrice = new BigDecimal(price);
+            lotResult.setLotTotalPrice(lotTotalPrice);
+        }
+    }
+
+    private void extractBuyer(JsonNode data, LotResult lotResult) {
+        Participant buyer = new Participant();
+        buyer.setName(data.get("procuringEntity").get("name").toString());
+        buyer.setEdrpou(data.get("procuringEntity").get("identifier").get("id").toString());
+        lotResult.setLotStatus(LotStatus.COMPLETED_PURCHASE);
+        if (checkSavedParticipant(buyer) != null) {
+            buyer = checkSavedParticipant(buyer);
+        } else {
+            participantRepository.save(buyer);
+        }
+        lotResult.setBuyer(buyer);
+    }
+
+    private void extractParticipants(JsonNode data, LotResult lotResult) {
+        JsonNode bids = data.get("bids");
+        if (bids != null) {
+            List<Participant> participants = new ArrayList<>();
+            for (JsonNode node : bids) {
+                Optional<JsonNode> tenderers = Optional.ofNullable(node.get("tenderers"));
+                if (tenderers.isPresent()) {
+                    for (JsonNode tenderer : tenderers.get()) {
+                        Participant participant = new Participant();
+                        participant.setName(tenderer.get("name").toString());
+                        participant.setEdrpou(tenderer.get("identifier").get("id").toString());
+                        if (checkSavedParticipant(participant) != null) {
+                            participant = checkSavedParticipant(participant);
+                        } else {
+                            participantRepository.save(participant);
+                        }
+                        participants.add(participant);
+                    }
+                }
```

Comment:
consider moving this code to a private method so this method doesn't have so many if and for steps